packages/@react-spectrum/button/src/Button.tsx

```diff
@@ -98,6 +98,61 @@ function Button<T extends ElementType = 'button'>(props: SpectrumButtonProps<T>,
     staticColor = 'white';
   }
 
+  /**
+   * Parse the children of the button to determine the text to return
+   * for the pending spinner aria-label.
+   * @returns A string that ends in a "pending", localized.
+   */
+  function getLabelFromChildren(): string {
+    // Text for spinner aria-label could come from one or more of:
+    // - child string
+    // - aria-label of button props
+    // - child Text component
+    // - child icon aria-label
+    // - could be absent
+    let label: string[] = [];
+
+    // string child
+    if (typeof children === 'string') {
+      label.push(children);
+    }
+
+    // aria-label on button
+    // Todo: Test how it announces in SRs
+    if (props['aria-label']) {
+      label.push(props['aria-label']);
+    }
+
+    if (React.isValidElement(children)) {
+      // Text wrapped in an component or element
+      if (typeof children?.props?.children === 'string') {
```

Comment:
likewise, can't do this because they may put many elements or components between the actual string or they may split the string with other elements `<i>The</i><span>button</span>`

Reply:
This case would be handled by line 138.